lib/src/main/java/org/altbeacon/beacon/startup/RegionBootstrap.java

```diff
@@ -34,24 +35,106 @@
+                monitorNotifier.didEnterRegion(region);
+            }
+
+            public void didExitRegion(Region region) {
+                monitorNotifier.didExitRegion(region);
+            }
+
+            public void didDetermineStateForRegion(int state, Region region) {
+                monitorNotifier.didDetermineStateForRegion(state, region);
+            }
+        };
+
+        regions = new ArrayList<Region>();
+        regions.add(region);
+        beaconConsumer = new InternalBeaconConsumer();
+        beaconManager.bind(beaconConsumer);
+        LogManager.d(TAG, "Waiting for BeaconService connection");
+    }
+
+    /**
+     * Constructor to bootstrap your Application on an entry/exit from multiple regions
+     *
+     * @param context
+     * @param regions
+     */
+    public RegionBootstrap(final Context context, final MonitorNotifier monitorNotifier, List<Region> regions) {
+        if (context.getApplicationContext() == null) {
+            throw new NullPointerException("Application Context should not be null");
+        }
+        beaconManager = BeaconManager.getInstanceForApplication(context.getApplicationContext());
+        this.context = context;
+        this.monitorNotifier = monitorNotifier;
+
+        this.monitorNotifier = new BootstrapNotifier() {
+            public Context getApplicationContext() {
+                return context.getApplicationContext();
+            }
+
+            public void didEnterRegion(Region region) {
+                monitorNotifier.didEnterRegion(region);
+            }
+
+            public void didExitRegion(Region region) {
+                monitorNotifier.didExitRegion(region);
+            }
+
+            public void didDetermineStateForRegion(int state, Region region) {
+                monitorNotifier.didDetermineStateForRegion(state, region);
+            }
+        };
+
+        this.regions = regions;
+
+        beaconConsumer = new InternalBeaconConsumer();
+        beaconManager.bind(beaconConsumer);
+        LogManager.d(TAG, "Waiting for BeaconService connection");
+    }
+
     /**
      * Constructor to bootstrap your Application on an entry/exit from a single region.
      *
      * @param application
      * @param region
+     * @deprecated This will be removed in a later release. Use
+     * * {@link org.altbeacon.beacon.startup.RegionBootstrap#RegionBootstrap(Context, MonitorNotifier, Region)}}
+     * * instead.
      */
     public RegionBootstrap(BootstrapNotifier application, Region region) {
         if (application.getApplicationContext() == null) {
             throw new NullPointerException("The BootstrapNotifier instance is returning null from its getApplicationContext() method.  Have you implemented this method?");
         }
         beaconManager = BeaconManager.getInstanceForApplication(application.getApplicationContext());
-        this.application = application;
+        this.context = application.getApplicationContext();
```

Comment:
if we consistently set this.context = application.getApplicationContext() in all constructors (which this appears to do), then all other places where we call this.context.getApplicationContext() can be changed to just this.context